zuul-core/src/main/java/com/netflix/zuul/passport/CurrentPassport.java

```diff
@@ -106,7 +107,7 @@ private Unlocker lock() {
     @VisibleForTesting
     public CurrentPassport(Ticker ticker) {
         this.ticker = ticker;
-        this.history = new LinkedList<>();
+        this.history = new ArrayDeque<>();
```

Comment:
I'm curious about this choice. I assumed the LinkedList was used to avoid needing to make copies when it grows

Reply:
The linter suggests that ArrayList/ArrayDeque almost always have better performance than LinkedList. Maybe due to locality?

Reply:
Makes sense, I could see the memory savings not being worth it